### PR TITLE
Fix ghcfilesystem target

### PR DESCRIPTION
**Pull request by uilianries**
Signed-off-by: Uilian Ries <uilianries@gmail.com>

Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ghcfilesystem/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -2,8 +2,10 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
```
Comment (prince-chrismc) [inline recipes/ghcfilesystem/all/test_package/CMakeLists.txt:5]: for my personal knowledge is the targets here required?

Comment (uilianries) [reply recipes/ghcfilesystem/all/test_package/CMakeLists.txt:5]: Yes, without TARGETS, conan_basic_setup will inject cmake module path, shadowing the target generated by cmake_find_package_multi.

It validates ghcfilesystem::ghcfilesystem if it's requesting all dependencies.

Actually it's an observation made by Madebr many time ago.

Comment (prince-chrismc) [reply recipes/ghcfilesystem/all/test_package/CMakeLists.txt:5]: Very interesting ! thanks!

**Comment by prince-chrismc** [comment]
@andioz you can merge this to unblock your PR

**Comment by andioz** [comment]
Hm, ok, I will do, but I have still open questions, I wanted to clarify this before merging...

**Comment by prince-chrismc** [comment]
I dont see any questions 🙊 sorry I was not aware!

**Comment by andioz** [comment]
It's merged now, and I added an updated question to the PR.

**Pull request merged**